app/src/app/services/connectors/rest-api/rest-api-connector.service.ts

```diff
@@ -406,6 +437,19 @@ export class RestApiConnectorService extends ApiConnector {
                         );
                     }
                 }),
+                switchMap(result => { // fetch parent data source of data component
+                    let x = result as any[];
+                    if (x[0].attackType != "data-component") return of(result);
+                    let d = x[0] as DataComponent;
+                    return this.getDataSource(d.data_source_ref).pipe( // fetch data source from REST API
+                        map(data_source => {
+                            let ds = data_source as DataSource[];
+                            d.data_source = ds[0];
+                            return [d];
+                        }),
+                        tap(result => logger.log("fetched data source of", result))
```

Comment:
`result` in this `tap` is duplicative name of `result` in the containing `switchMap`.